File: backend/Service/Services/CategoryService.cs
using Data.Dtos;
using Data.Interfaces;
using Service.Interfaces;

namespace Service.Services;

public class CategoryService:  ICategoryService
{
    private readonly ICategoryRepository _categoryRepository;
    
    public CategoryService(ICategoryRepository categoryRepository)
    {
        _categoryRepository = categoryRepository;
    }
    
    public async Task<List<CategoryDto>> GetCategories()
    {
        return await _categoryRepository.GetCategories();
    }

    public async Task<CategoryDto?> GetCategoryById(int categoryId)
    {
        return await _categoryRepository.GetCategoryById(categoryId);
    }

    public async Task<CategoryDto?> AddCategory(CreateCategoryDto newCategory)
    {
        if (newCategory.CategoryName == "") return null;
        return await _categoryRepository.AddCategory(newCategory);
    }

    public async Task<bool> UpdateCategory(int categoryId, CreateCategoryDto updatedCategory)
    {
        if (updatedCategory.CategoryName == "") return false;
        return await _categoryRepository.UpdateCategory(categoryId, updatedCategory);
    }

    public async Task DeleteCategory(int categoryId)
    {
        await _categoryRepository.DeleteCategory(categoryId);
    }
}

File: frontend/src/services/CategoryService.ts
import api from "./api";

export const getCategories = async () => {
    const response = await api.get("/category");
    return response.data;
}

export const getCatgoryById = async (categoryId: number) => {
    const response = await api.get(`/category/${categoryId}`);
    return response.data;
}

export const addCategory = async (categoryName: string) => {
    var newCategory = {
        CategoryName: categoryName
    }
    const response = await api.post("/category/add", newCategory);
    return response;
}

export const deleteCategoryById = async (categoryId: number) => {
    const response = await api.put(`/category/delete/${categoryId}`);
    return response;
}

export const updateCategory = async(categoryId: number, categoryName: string) => {
    var updateCategory = {
        CategoryName: categoryName
    }
    const response = await api.put(`/category/update/${categoryId}`, updateCategory);
    return response;
}